units/systemd-logind.service.in

```diff
@@ -31,7 +31,9 @@ DeviceAllow=char-input rw
 DeviceAllow=char-tty rw
 DeviceAllow=char-vcs rw
 ExecStart={{LIBEXECDIR}}/systemd-logind
-FileDescriptorStoreMax=512
+FileDescriptorStoreMax=768
+# We want to be able to pick up session leader fds even after a stop.
```

Comment:
do we? can you elaborate?

usually we should set FileDescriptorStorePreserve= only for thigns that are relevat for soft reboot. But this stuff isn't really?

Reply:
In both TEST-35-LOGIN and debian autopkgtest, stopping logind completely and then start it is a valid test case. So I consider that supported?

Reply:
well, kinda. but "supported" is a relative term. keeping the fdstore around for good will probably piss of people who will be surprised that DRM devices and input devices and such remain opened by logind even though logind has died.

the way i see it: the drawbacks of keeping the fdstore for resources around that aren't just ours (i.e. an fd to a drm or input device is a system resource we keep pin, which is different from a memfd we create ourselves, which is pretty much just ours) is too large to just enabled this by default.

I mean, we have a reasonable logic in place that we can be stopped and later be restarted also without pidfd being uploaded into the fdstore, we can just use the PID after all, which usually is good enough, even if racy. But stopping and starting logind at arbitrar ytimes is not a clean code path anyway, it's already messy to do this, hence I don't think we need to cover for this case.

Hence: if "systemctl restart systemd-logind" keeps the pidfds around, that's a usecase we actively want to support, and it's good. otoh also doing this on "systemctl stop systemd-logind && systemctl start systemd-logind" seems to be too much to ask and comes with too many drawbacks, and is an exotic usecase.

if that makes any sense?

hence i would drop the FileDescriptorStorePreserve= line

Reply:
I'm OK with that. However, the test case would need to be removed (from Debian autopkgtest too)

Reply:
maybe in the test jus tadd the drop-in that sets FileDescriptorStorePreserve=?

also, the test, doesn't that also work if it uses plain pids rather than pidfds?

Reply:
> doesn't that also work if it uses plain pids rather than pidfds?

The current implementation in this PR doesn't fall back when `leader_fd_saved` is true. We just drop (gc) the session if pidfd is lost.

Reply:
IMO it's better to just drop the test instead of patching over it further. The session devices are lost no matter what, so the state of sessions after a coldplug is ill-defined. But I really don't know how to deal with autopkgtest...

Reply:
cc @bluca: The test case was there ever since the autopkgtest was introduced. But as discussed, maybe it should be removed 🤔

Reply:
Only speculating, but I imagine it was added because a long time ago the packaging script didn't do `systemctl restart foo` on package upgrades, but `systemctl stop foo` and then later `systemctl start foo`. This is no longer the case, however, using a drop-in and keeping the test seems like a good idea to me, so we exercise those corner cases

Reply:
Okay. Added.